Support for S3 object tagging

jlahd opened the pull request.

xach commented:
Hi, this has a couple issues.

First, please use spaces instead of tabs for indentation.

Second, the documentation must also be updated to describe the new feature.

Third, it appears as though the feature has to do with a "tagging" option, but the parameter in code is "tags" - could these be made more uniform?

Can you also please add, in a comment or similar, a link or reference to the applicable S3 documentation for tagging?

Thanks!

jlahd commented:
Hello, thanks for the comments!

1. Sorry - I thought I had untabified the source, but apparently not. Fixed.

2. Done.

3. In S3 documentation, the terms are used in quite mixed fashion. Generally, it seems that the feature is called "tagging", while the single key-value pairs are called "tags". Changed now to "tagging".

Also, I added three more functions (get-tagging, put-tagging, delete-tagging) to complement the functionality related to tagging. The links to S3 documentation can be found in the documentation changes.
